PIPReader/src/main/java/it/cnr/iit/pipreader/PIPReader.java

```diff
@@ -90,27 +90,32 @@ public PIPReader( PipProperties properties ) {
         log.info( "PIPReader " + properties.getId() + " initialised" );
     }
 
+    // TODO use reject here
     private boolean init( PipProperties properties ) {
         try {
             Map<String, String> attributeMap = properties.getAttributes().get( 0 );
             Attribute attribute = new Attribute();
 
-            if( !attribute.createAttributeId( attributeMap.get( ATTRIBUTE_ID ) ) ) {

```

Comment:
the attribute class was using a weird create prefix returning boolean. instead of the common void set prefix to change the attributeId.
it has been replaced with this : attribute.setAttributeId( attributeId ); that could triggere the reject class

Reply:
well! 
since it uses Reject we actually don't need the init to return a boolean or to do those checks and logs